File: Programmers/javascript/hanghae99_4th/q44_최대공약수와_최소공배수.js
function solution(n, m) {
    var answer = [];
    const gcd = getGCD(n, m);
    answer.push(gcd); // gcd
    answer.push((n * m) / gcd); // lcm
    return answer;
}

function getGCD(n, m) {
    // 유클리드 호제법
    if (n % m === 0) return m;
    else return getGCD(m, n % m);
}

File: Programmers/python/prac/q44_최대공약수와_최소공배수.py
def solution(n, m):
    answer = []
    gcd = get_gcd(n, m)
    answer.append(gcd)  # gcd
    answer.append(n*m/gcd)  # lcm
    return answer

def get_gcd(n, m):  # 유클리드 호제법
    return m if n % m == 0 else get_gcd(m, n % m)
